### PR TITLE
Adding the cli to be published to Docker hub

### PR DESCRIPTION
**Pull request by josmo**
@bradrydzewski Let me know if this works for you or if I should find another way to publish to docker hub :)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# docker build --rm -t drone/drone .
```
Comment (appleboy) [inline Dockerfile]: `drone/drone-cli` ?

File: .drone.yml [hunk 2]
```diff
@@ -24,3 +24,11 @@ pipeline:
...
     when:
       event: tag
+
+  docker_publish:
+    image: plugins/docker
+    repo: drone/drone-cli
+    secrets: [ docker_username, docker_password ]
+    tag: [ latest ]
+    when:
+      event: tag
```
Comment (appleboy) [inline .drone.yml]: Separate two steps:

```yml
  publish_latest:
    image: plugins/docker
    repo: ${DRONE_REPO}
    tags: [ 'latest' ]
    secrets: [ docker_username, docker_password ]
    when:
      event: [ push ]
      branch: [ master ]

  publish_tag:
    image: plugins/docker
    repo: ${DRONE_REPO}
    tags: [ '${DRONE_TAG}' ]
    secrets: [ docker_username, docker_password ]
    group: release
    when:
      event: [ tag ]
      branch: [ refs/tags/* ]
```

File: .drone.yml [hunk 3]
```diff
@@ -24,3 +24,22 @@ pipeline:
         target: github_release_api_key
     when:
       event: tag
+
+ publish_latest:
```
Comment (tboerger) [inline .drone.yml]: I think this is not indented correctly.

File: .drone.yml [hunk 4]
```diff
@@ -24,3 +24,22 @@ pipeline:
...
+    image: plugins/docker
+    repo: ${DRONE_REPO}
```
Comment (tboerger) [inline .drone.yml]: Why this variable? I have never seen that in any other drone repo.

File: .drone.yml (hunk 3 above)
Comment (josmo) [inline .drone.yml]: oh oops, guess that's what happens when I hurry with vim :) I'll have that fix in a sec

File: .drone.yml (hunk 4 above)
Comment (josmo) [inline .drone.yml]: it's the project/repo in this case 'drone/drone-cli'

**Comment by intellectronica** [comment]
This will be extremely useful for me (currently using a similar image that I roll). When will this be merged?

File: .drone.yml [hunk 5]
```diff
@@ -24,3 +24,22 @@ pipeline:
...
+    when:
+      event: [ push ]
+      branch: [ master ]
+
+  publish_tag:
+    image: plugins/docker
+    repo: ${DRONE_REPO}
+    tags: [ '${DRONE_TAG}' ]
+    secrets: [ docker_username, docker_password ]
+    group: release
```
Comment (tboerger) [inline .drone.yml]: This `group` doesn't make sense

File: .drone.yml [hunk 6]
```diff
@@ -24,3 +24,22 @@ pipeline:
         target: github_release_api_key
     when:
       event: tag
+
+  publish_latest:
```
Comment (tboerger) [inline .drone.yml]: What about just calling both `docker`? It can anyway just execute one of them.

File: .drone.yml (hunk 5 above)
Comment (appleboy) [inline .drone.yml]: yes. please remove `group`.

Comment (josmo) [inline .drone.yml]: done

**Comment by josmo** [comment]
@bradrydzewski didn't know if you decided to include this or not :)

**Comment by Tim15** [comment]
This is a very useful feature for me. I'd love to just run 

```
$ docker run -it --env-file=drone.env drone/cli drone repo ls
```

Instead of downloading the binaries from github. Drone is container-centric after all!

Seems simple and ready to merge. Are you planning to add this? Ping @bradrydzewski.

File: .drone.yml (hunk 6 above)
Comment (tboerger) [reply .drone.yml]: Call it both docker or publish

File: .drone.yml [hunk 7]
```diff
@@ -24,3 +24,21 @@ pipeline:
...
+    image: plugins/docker
+    repo: ${DRONE_REPO}
```
Comment (tboerger) [inline .drone.yml]: Don't use this variable, make it more explicit

File: .drone.yml [hunk 8]
```diff
@@ -24,3 +24,21 @@ pipeline:
...
+    tags: [ 'latest' ]
+    secrets: [ docker_username, docker_password ]
+    when:
+      event: [ push ]
+      branch: [ master ]
+
+  publish_tag:
+    image: plugins/docker
+    repo: ${DRONE_REPO}
+    tags: [ '${DRONE_TAG}' ]
```
Comment (tboerger) [inline .drone.yml]: Same here

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# docker build --rm -t drone/drone-cli .
+FROM centurylink/ca-certs
```
Comment (tboerger) [inline Dockerfile]: Use the drone ca-certs image

**Comment by josmo** [comment]
@tboerger I think this should have all your comments covered. I also parallelized the builds.

File: .drone.yml [hunk 10]
```diff
@@ -4,17 +4,80 @@ workspace:
...
+    environment:
+      - GOOS=linux
+      - GOARCH=arm
+      - CGO_ENABLED=0
+    commands:
+      - go build -v -ldflags "-X main.version=${DRONE_TAG##v}" -a -o release/darwin/amd64/drone github.com/drone/drone-cli/drone
+      - tar -cvzf release/drone_darwin_amd64.tar.gz  -C release/darwin/amd64  drone
 
-  publish:
+  generate_shas:
```
Comment (tboerger) [inline .drone.yml]: You can skip this and activate the checksums within the github release plugin

File: .drone.yml [hunk 11]
```diff
@@ -4,17 +4,80 @@ workspace:
...
+    environment:
+      - GOOS=linux
+      - GOARCH=arm
+      - CGO_ENABLED=0
+      - GOARM=7
+    commands:
+      - go build -v -ldflags "-X main.version=${DRONE_TAG##v}" -a -o release/linux/arm/drone github.com/drone/drone-cli/drone
+      - tar -cvzf release/drone_linux_arm.tar.gz     -C release/linux/arm     drone
+
+  build_windows:
```
Comment (tboerger) [inline .drone.yml]: And if we are really picky this should also get an _amd64 :P

File: Dockerfile [hunk 12]
```diff
@@ -0,0 +1,11 @@
+FROM plugins/base:multiarch
```
Comment (tboerger) [inline Dockerfile]: I think brad will prefer drone/ca-certs

File: Dockerfile [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+MAINTAINER Drone.IO Community <drone-dev@googlegroups.com>
```
Comment (tboerger) [inline Dockerfile]: @bradrydzewski ok with that? Or Drone.IO Inc.?

File: .drone.yml (hunk 11 above)
Comment (josmo) [reply .drone.yml]: lol yeah probably right

File: .drone.yml (hunk 10 above)
Comment (josmo) [reply .drone.yml]: oh had no idea. I'll switch it :)

File: .drone.yml [hunk 14]
```diff
@@ -4,23 +4,114 @@ workspace:
...
+    environment:
+      - GOOS=windows
+      - GOARCH=amd64
+      - CGO_ENABLED=0
+    commands:
+      - go build -v -ldflags "-X main.version=${DRONE_TAG##v}" -a -o release/windows/adm64/drone github.com/drone/drone-cli/drone
+      - tar -cvzf release/drone_windows_amd64.tar.gz -C release/windows/amd64 drone
 
-  publish:
+  build_darwin:
```
Comment (tboerger) [inline .drone.yml]: Also amd64... hrhr

Comment (josmo) [reply .drone.yml]: grr thanks for catching that, I also had a stupid error for the GOOS and GOARCH.

**Comment by bradrydzewski** [comment]
I think the scope of this was a bit too large. I use slightly different conventions in the drone organization, and this pull request broke with the conventions we're using in the drone/drone project.

I pushed d7b19e6f6f581880352f232ba3453e0b769a9dca which we can use as a baseline going forward.

**Pull request closed without merging**